controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -343,31 +377,37 @@ public void processingFailure(Exception error) {
 
             @Override
             public void connectionDropped() {
+                log.warn("abortTransaction {}/{}/{} connectionDropped", scope, stream, segmentNumber);
                 result.completeExceptionally(new WireCommandFailedException(type, WireCommandFailedException.Reason.ConnectionDropped));
             }
 
             @Override
             public void wrongHost(WireCommands.WrongHost wrongHost) {
+                log.warn("abortTransaction {}/{}/{} wrongHost", scope, stream, segmentNumber);
                 result.completeExceptionally(new WireCommandFailedException(type, WireCommandFailedException.Reason.UnknownHost));
             }
 
             @Override
             public void transactionCommitted(WireCommands.TransactionCommitted transactionCommitted) {
+                log.warn("abortTransaction {}/{}/{} TransactionCommitted", scope, stream, segmentNumber);
```

Comment:
This one does not sound like a WARN.

Reply:
it is warn because this is a failure case, we are trying to abort a transaction that is already committed.

Reply:
though note: controller should never get into this state.

Reply:
ok